### ISSUE TITLE
Bug: Changing the API and regenerating client-go does not add changes to code repo.

### ISSUE THREAD

**Issue by kannon92** [kubernetes-sigs/jobset#219]
When I modify the api and try to generate client-go, my client-go is generated and placed in my go src directory (ie /home/ec2-user/go/1.20.2/src/sigs.k8s.io/jobset/client-go/).  

I think that it should go to client-go folder in the repo.  Currently if I modify the API I would have to copy /home/ec2-user/go/1.20.2/src/sigs.k8s.io/jobset/client-go/ and rewrite the folder in client-go.  

This is not ideal.  

Any ideas on what is going wrong?

**Comment by danielvegamyhre** [comment]
When originally developing this I found that `generate-groups.sh` script called by `update-codegen.sh` accepts an "output package" parameter, which I specified as `sigs.k8s.io/jobset/client-go`, and treats each thing between the `/` chars as a directory relative to `$GOROOT/src`. So in my case this writes to `~/go/src/sigs.k8s.io/jobset/client-go`, and since my jobset repo is located at `~/go/src/sigs.k8s.io/jobset`, this is the correct location.

However, I can reproduce this issue by cloning jobset in my /tmp directory and running `make`, it still writes the client-go to the same location relative to my Go root, not relative to my current working directory. I don't see an obvious way to change this in the script, as it only seems to accept an "output package" parameter, not a filesystem location of to create output files relative to.

I think maybe at the end of the script we just need to move the output `client-go` directory to the location relative to the current working directory.
